src/Native/Unix/System.Native/pal_interfaceaddresses.cpp

```diff
@@ -44,6 +44,12 @@ extern "C" int32_t SystemNative_EnumerateInterfaceAddresses(IPv4AddressFound onI
     for (ifaddrs* current = headAddr; current != nullptr; current = current->ifa_next)
     {
         uint32_t interfaceIndex = if_nametoindex(current->ifa_name);
+        // ifa_name may be an aliased interface name.
+        // Use if_indextoname to map back to the true device name.
+        char actualName[IF_NAMESIZE];
+        char* result = if_indextoname(interfaceIndex, actualName);
+        assert(result == actualName && result != nullptr);
```

Comment:
> For the specific case you mention

Alternatively, if that error would only occur in a race condition like that, we could simply retry in a loop.  The chances that it would happen multiple times are probably miniscule; if we were really concerned, we could throw after N attempts.